enhancements/machine-api/aws-placement-groups.md

```diff
@@ -0,0 +1,543 @@
+labels.
+
+For example, across three MachineSets, the Placement would vary only by the partition number:
+
+```yaml
+providerSpec:
+  value:
+    placement:
+      region: us-east-1
+      availabilityZone: us-east-1a
+      groupName: my-partition-placement-group
+      groupType: Partition
+      partitionCount: 3
+      partitionNumber: 1 | 2 | 3 # This value will vary across the three MachineSets, either 1, 2 or 3
+```
+
+###### Limitations of Partition Placement Groups
+
+- A partition placement group with AWS Dedicated Instances can have a maximum of two partitions.
+
+#### Creation of Placement Groups
+
+To enable users to get started with Placement Groups, we will allow users to specify the name of a placement group via
+the new `placement.groupName` field on their MachineSet, which will then be used for the placement of new Machines
+created by the Machineset.
+If the placement group does not exist, we will create it based on the configuration provided by the user.
+
+We will allow the user to specify the type of the placement group to create, either Cluster, Spread or Partition, as
+well as, in the case of Partition, the number of partitions to be created.
+
+Should the user not define the type, we will return an error and place the Machine into a Failed state. Due to the
+differences in behaviour between the different placement group strategies, we do not believe there is a sensible default
+choice and as such, the group type will be required when creating a placement group.
+
+Should the user not define the number of partitions, we will omit this value in the request and leave AWS to default
+the value.
+Currently the default behaviour is to create 2 partitions if otherwise unspecified.
+
+When creating the placement group, we will specify the `Name` tag as well as the standard cluster ID tag that is used
+to identify resources belonging to a particular cluster.
+
+As group names must be unique within a region, we should not fall foul of eventual consistency issues leading to
+leaking of placement groups as we can do with EC2 instances.
+
+#### Removal of Placement Groups
+
+To ensure we leave user accounts clean, we should remove any placement group that is no longer required.
+
+Each time a Machine is removed, we must check whether it belongs to a placement group.
+If it belongs to a placement group, we must also check that there are no other instances within the placement group.
+This will mean listing all instances in the region, filtering them down based on their placement group name.
+
+We should be able to replicate the following CLI command within the AWS Machine controller for this purpose:
+```shell
+aws ec2 describe-instances --filters Name=placement-group-name,Values=<placement-group-name>
+```
+
+If no other instance exists within the group, we will also delete the placement group once the instance has been
+removed.
```

Comment:
If you try to delete a placement group that has an instance within it, AWS will return a bad request, this would then likely cause the controller to requeue that machine request, at which point we would see there's an instance in the group now (I would expect) and skip trying to delete the PG